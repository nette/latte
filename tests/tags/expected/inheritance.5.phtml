<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	protected const BLOCKS = [
		['content' => 'blockContent'],
	];


	public function main(): array
	{
%A%
		if ($this->getParentName()) {
			return get_defined_vars();
		}
		$this->renderBlock('content', get_defined_vars()) /* line %d% */;
%A%
	}


	public function prepare(): void
	{
		extract($this->params);
		$this->parentName = (true ? $ext : "undefined");
	}


	/** {block content} on line %d% */
	public function blockContent(array $ÊŸ_args): void
	{
		echo '	Content
';
	}

}
