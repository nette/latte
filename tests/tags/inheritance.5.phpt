<?php

/**
 * Test: {extends ...} test V.
 */

declare(strict_types=1);

use Tester\Assert;


require __DIR__ . '/../bootstrap.php';



$latte = new Latte\Engine;
$latte->setLoader(new Latte\Loaders\StringLoader([
	'parent' => file_get_contents(__DIR__ . '/templates/parent.latte'),

	'main' => <<<'XX'

		{extends true ? $ext : "undefined"}

		{block content}
			Content
		{/block}

		XX,
]));

Assert::matchFile(
	__DIR__ . '/expected/inheritance.5.phtml',
	$latte->compile('main'),
);
Assert::matchFile(
	__DIR__ . '/expected/inheritance.5.html',
	$latte->renderToString('main', ['ext' => 'parent']),
);
