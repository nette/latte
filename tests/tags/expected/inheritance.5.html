<!DOCTYPE html>
<head>
	<title>MY WEBSITE</title>
</head>

<body>
	<div id="sidebar">
		<ul>
			<li><a href="/">Homepage</a></li>
		</ul>
	</div>

	<div id="content">
	Content

	CONTENT
	</div>
</body>
</html>
Parent: main/extends
